pkg/network/dhcp/dhcp.go

```diff
@@ -38,6 +40,45 @@ func (d *Probe) Start(inf string) error {
 	return nil
 }
 
+// Run runs the DHCP client process and registers it to zinit
+func (d *Probe) Run(inf string) error {
+	z, err := zinit.New("")
+	if err != nil {
+		log.Error().Err(err).Msg("failed to connect to zinit")
+		return err
+	}
+
+	serviceName := fmt.Sprintf("dhcp-%s", inf)
+
+	ns, err := exec.Command("ip", "netns", "identify").Output()
+	if err != nil {
+		return errors.Wrap(err, "failed to identify namespace")
+	}
+
+	err = zinit.AddService(serviceName, zinit.InitService{
+		Exec:    fmt.Sprintf("ip netns exec %s /sbin/udhcpc -v -f -i %s -t 20 -T 1 -s /usr/share/udhcp/simple.script", ns, inf),
+		Oneshot: false,
+		After:   []string{},
+	})
+
+	if err != nil {
+		log.Error().Err(err).Msg("fail to create dhcp-zos zinit service")
+		return err
+	}
+
+	if err := z.Monitor(serviceName); err != nil {
+		log.Error().Err(err).Msg("fail to start monitoring dhcp-zos zinit service")
+		return err
+	}
+
+	if err := z.Start(serviceName); err != nil {
```

Comment:
zinit monitor already auto starts the service. You do Start only if you did a Stop before. You can check zinit docs for more details about the serivce live cycle.